src/main/java/io/quarkus/fs/util/ZipUtils.java

```diff
@@ -114,20 +114,31 @@ public static void zip(Path src, Path zipFile) throws IOException {
     }
 
     public static FileSystem newZip(Path zipFile) throws IOException {
-        final Map<String, Object> env;
+        return newZip(zipFile, Collections.emptyMap());
+    }
+
+    /**
+     * In this version, callers can set whatever parameters they like for the FileSystem.
+     * Note that these values always take precedence over whatever the library itself sets.
+     */
+    public static FileSystem newZip(Path zipFile, Map<String, Object> env) throws IOException {
+        final Map<String, Object> effectiveEnv;
         if (Files.exists(zipFile)) {
-            env = DEFAULT_OWNER_ENV;
+            effectiveEnv = new HashMap<>(DEFAULT_OWNER_ENV);
         } else {
-            env = CREATE_ENV;
+            effectiveEnv = new HashMap<>(CREATE_ENV);
             // explicitly create any parent dirs, since the ZipFileSystem only creates a new file
             // with "create" = "true", but doesn't create any parent dirs.
 
             // It's OK to not check the existence of the parent dir(s) first, since the API,
             // as per its contract doesn't throw any exception if the parent dir(s) already exist
             Files.createDirectories(zipFile.getParent());
         }
+        if (env != null) {
+            effectiveEnv.putAll(env);
```

Comment:
How about creating a copy only when env isn't null and isn't empty?

Reply:
Makes sense